### ISSUE TITLE
Maven 3.9.2 warnings about compatibility with future Maven versions

### ISSUE THREAD

**Issue by xak2000** [aleksandr-m/gitflow-maven-plugin#394]
Maven 3.9.2 complains about `gitflow-maven-plugin` 1.20.0:

```
[WARNING] 
[WARNING] Plugin validation issues were detected in 1 plugin(s)
[WARNING] 
[WARNING]  * com.amashchenko.maven.plugin:gitflow-maven-plugin:1.20.0
[WARNING]   Declared at location(s):
[WARNING]    * com.example:demo:0.0.1-SNAPSHOT (pom.xml) @ line 38
[WARNING]   Used in module(s):
[WARNING]    * com.example:demo:0.0.1-SNAPSHOT (pom.xml)
[WARNING]   Plugin issue(s):
[WARNING]    * Plugin is a Maven 2.x plugin, which will be not supported in Maven 4.x
[WARNING]    * Plugin mixes multiple Maven versions: [3.8.5, 2.2.1]
[WARNING]    * Plugin should declare these Maven artifacts in `provided` scope: [org.apache.maven:maven-repository-metadata:3.8.5, org.apache.maven:maven-artifact:3.8.5, org.apache.maven:maven-settings-builder:3.8.5, org.apache.maven:maven-resolver-provider:3.8.5, org.apache.maven:maven-model-builder:3.8.5, org.apache.maven:maven-core:3.8.5, org.apache.maven:maven-plugin-api:3.8.5, org.apache.maven:maven-model:3.8.5, org.apache.maven:maven-builder-support:3.8.5, org.apache.maven:maven-settings:3.8.5]
[WARNING]    * Plugin depends on plexus-container-default, which is EOL
[WARNING] 
[WARNING] 
[WARNING] Fix reported issues by adjusting plugin configuration or by upgrading above listed plugins. If no upgrade available, please notify plugin maintainers about reported issues.
[WARNING] For more or less details, use 'maven.plugin.validation' property with one of the values (case insensitive): [BRIEF, DEFAULT, VERBOSE]
[WARNING] 
```

See Maven 3.9.2 [Release Notes](https://maven.apache.org/docs/3.9.2/release-notes.html#notable-new-features) - the part about "Plugin validation warnings change".

### Steps to reproduce:
* Use Maven 3.9.2
* `mvn gitflow:help -Dmaven.plugin.validation=VERBOSE`

Probably related to #369, #389.

**Comment by xak2000** [comment]
One thing could be done without updating the dependencies:

> Plugin should declare these Maven artifacts in `provided` scope:
* org.apache.maven:maven-repository-metadata:3.8.5
* org.apache.maven:maven-artifact:3.8.5
* org.apache.maven:maven-settings-builder:3.8.5
* org.apache.maven:maven-resolver-provider:3.8.5
* org.apache.maven:maven-model-builder:3.8.5org.apache.maven:maven-core:3.8.5, org.apache.maven:maven-plugin-api:3.8.5, org.apache.maven:maven-model:3.8.5, org.apache.maven:maven-builder-support:3.8.5, org.apache.maven:maven-settings:3.8.5]

`provided` scope

**Comment by DRoppelt** [comment]
Just verified 1.19.0 vs 1.20.0 (just released) vs 1.20.1-SNAPSHOT (with changes of PR #389)

Looks like all warnings get solved with that PR

`1.19.0`

```
$ mvn com.amashchenko.maven.plugin:gitflow-maven-plugin:1.19.0:help -D maven.plugin.validation=VERBOSE
...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.276 s
[INFO] Finished at: 2023-06-13T22:09:43+02:00
[INFO] ------------------------------------------------------------------------
[WARNING]
[WARNING] Plugin validation issues were detected in 1 plugin(s)
[WARNING]
[WARNING]  * com.amashchenko.maven.plugin:gitflow-maven-plugin:1.19.0
[WARNING]   Declared at location(s):
[WARNING]    * unknown
[WARNING]   Used in module(s):
[WARNING]    * com.amashchenko.maven.plugin:gitflow-maven-test:0.0.3-test-SNAPSHOT (pom.xml)
[WARNING]   Plugin issue(s):
[WARNING]    * Plugin is a Maven 2.x plugin, which will be not supported in Maven 4.x
[WARNING]    * Plugin mixes multiple Maven versions: [3.0-alpha-2, 3.3.9, 2.2.1]
[WARNING]    * Plugin should declare these Maven artifacts in `provided` scope: [org.apache.maven:maven-settings-builder:3.3.9, org.apache.maven:maven-artifact:3.3.9, org.apache.maven:maven-builder-support:3.3.9, org.apache.maven:maven-project-builder:3.0-alpha-2, org.apache.maven:maven-settings:3.3.9, org.apache.maven:maven-compat:3.0-alpha-2, org.apache.maven:maven-project:3.0-alpha-2, org.apache.maven:maven-model-builder:3.3.9, org.apache.maven:maven-repository-metadata:3.3.9, org.apache.maven:maven-core:3.3.9, org.apache.maven:maven-aether-provider:3.3.9, org.apache.maven:maven-plugin-api:3.3.9, org.apache.maven:maven-model:3.3.9]
[WARNING]    * Plugin depends on plexus-container-default, which is EOL
[WARNING]
[WARNING]
[WARNING] Fix reported issues by adjusting plugin configuration or by upgrading above listed plugins. If no upgrade available, please notify plugin maintainers about reported issues.
[WARNING] For more or less details, use 'maven.plugin.validation' property with one of the values (case insensitive): [BRIEF, DEFAULT, VERBOSE]
[WARNING]
```

`1.20.0`
```
$ mvn com.amashchenko.maven.plugin:gitflow-maven-plugin:1.20.0:help -D maven.plugin.validation=VERBOSE
...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.293 s
[INFO] Finished at: 2023-06-13T22:09:55+02:00
[INFO] ------------------------------------------------------------------------
[WARNING]
[WARNING] Plugin validation issues were detected in 1 plugin(s)
[WARNING]
[WARNING]  * com.amashchenko.maven.plugin:gitflow-maven-plugin:1.20.0
[WARNING]   Declared at location(s):
[WARNING]    * unknown
[WARNING]   Used in module(s):
[WARNING]    * com.amashchenko.maven.plugin:gitflow-maven-test:0.0.3-test-SNAPSHOT (pom.xml)
[WARNING]   Plugin issue(s):
[WARNING]    * Plugin is a Maven 2.x plugin, which will be not supported in Maven 4.x
[WARNING]    * Plugin mixes multiple Maven versions: [3.8.5, 2.2.1]
[WARNING]    * Plugin should declare these Maven artifacts in `provided` scope: [org.apache.maven:maven-repository-metadata:3.8.5, org.apache.maven:maven-artifact:3.8.5, org.apache.maven:maven-settings-builder:3.8.5, org.apache.maven:maven-resolver-provider:3.8.5, org.apache.maven:maven-model-builder:3.8.5, org.apache.maven:maven-core:3.8.5, org.apache.maven:maven-plugin-api:3.8.5, org.apache.maven:maven-model:3.8.5, org.apache.maven:maven-builder-support:3.8.5, org.apache.maven:maven-settings:3.8.5]
[WARNING]    * Plugin depends on plexus-container-default, which is EOL
[WARNING]
[WARNING]
[WARNING] Fix reported issues by adjusting plugin configuration or by upgrading above listed plugins. If no upgrade available, please notify plugin maintainers about reported issues.
[WARNING] For more or less details, use 'maven.plugin.validation' property with one of the values (case insensitive): [BRIEF, DEFAULT, VERBOSE]
[WARNING]
```

`1.20.1-SNAPSHOT`
```
$ mvn com.amashchenko.maven.plugin:gitflow-maven-plugin:1.20.1-SNAPSHOT:help -D maven.plugin.validation=VERBOSE
...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.065 s
[INFO] Finished at: 2023-06-13T22:10:04+02:00
[INFO] ------------------------------------------------------------------------
```
